operations/producer/producer-operation.go

```diff
@@ -91,25 +119,33 @@ func (operation *ProducerOperation) Produce(topic string, flags ProducerFlags) {
 	if flags.Value != "" {
 		value = []byte(flags.Value)
 	} else if stdinAvailable() {
-		bytes, err := ioutil.ReadAll(os.Stdin)
-		if err != nil {
-			output.Failf("Failed to read data from the standard input: %s", err)
-		}
+		scanner := bufio.NewScanner(os.Stdin)
+		for scanner.Scan() {
+			bytes, err := scanner.Text(), scanner.Err()
+			if err != nil {
+				output.Failf("Failed to read data from the standard input: %s", err)
+			}
 
-		if len(bytes) > 0 && bytes[len(bytes)-1] == 10 {
-			// remove trailing return
-			bytes = bytes[:len(bytes)-1]
-		}
+			if len(bytes) > 0 && bytes[len(bytes)-1] == 10 {
+				// remove trailing return
+				bytes = bytes[:len(bytes)-1]
+			}
 
-		if flags.Separator != "" {
-			input := strings.SplitN(string(bytes[:]), flags.Separator, 2)
-			if len(input) < 2 {
-				output.Failf("the provided input does not contain the separator %s", flags.Separator)
+			if flags.Separator != "" {
+				// In this case we need to buffer the requests or use another seperator to seperate messages
```

Comment:
I don't get what you mean. If a separator is provided, this separator is used to divide a single line into two parts (message-key, message-value). so I think it is ok like this

Reply:
left it as is, thanks (ill later delete the comment)